File: lib/features/storage/domain/history_repository.dart
import 'models/history_entry.dart';

/// Репозиторий истории распознаваний (§4.5 ТЗ).
///
/// Хранит последние 50 записей в SQLite.
/// Изображения НЕ сохраняются по умолчанию.
abstract class HistoryRepository {
  /// Добавляет запись в историю.
  ///
  /// Если записей > 50, удаляет самую старую.
  Future<void> addEntry(HistoryEntry entry);

  /// Возвращает последнюю запись (для команды «Повтори»).
  Future<HistoryEntry?> getLastEntry();

  /// Возвращает все записи (для экрана истории).
  Future<List<HistoryEntry>> getAllEntries();

  /// Очищает всю историю (по запросу пользователя).
  Future<void> clearHistory();
}

File: src/features/storage/domain/historyRepository.ts
/**
 * Интерфейс репозитория истории распознаваний.
 * Лимит: HISTORY_MAX_ENTRIES (50) записей.
 */

import { HistoryEntry } from './historyEntry';

export interface HistoryRepository {
  /** Добавить запись. Если превышен лимит — удалить самую старую. */
  addEntry(entry: Omit<HistoryEntry, 'id'>): Promise<void>;

  /** Получить последнюю запись (для команды «Повтори») */
  getLastEntry(): Promise<HistoryEntry | null>;

  /** Получить все записи (новые первыми) */
  getAllEntries(): Promise<HistoryEntry[]>;

  /** Очистить всю историю */
  clearHistory(): Promise<void>;
}
